deploy/awstools/awstools.py

```diff
@@ -55,7 +62,76 @@ def valid_aws_configure_creds():
         return False
     return True
 
-def aws_resource_names():
+def get_localhost_instance_info(url_ext: str) -> Optional[str]:
+    """ Obtain latest instance info from instance metadata service. See
+    https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+    for more info on what can be accessed.
+
+    Args:
+        url_ext: Part of URL after 169.254.169.254/latest
+
+    Returns:
+        Data obtained in string form or None
+    """
+    res = None
+    # This takes multiple minutes without a timeout from the CI container. In
+    # practise it should resolve nearly instantly on an initialized EC2 instance.
+    curl_connection_timeout = 10
+    with settings(ok_ret_codes=[0,28]), hide('everything'):
+        res = local(f"curl -s --connect-timeout {curl_connection_timeout} http://169.254.169.254/latest/{url_ext}", capture=True)
+        rootLogger.debug(res.stdout)
+        rootLogger.debug(res.stderr)
+
+    if res.return_code == 28:
+        return None
+    else:
+        return res.stdout
+
+def get_localhost_instance_id() -> Optional[str]:
+    """Get current manager instance id, if applicable.
+
+    Returns:
+        A ``str`` of the instance id or ``None``
+    """
+
+    return get_localhost_instance_info("meta-data/instance-id")
+
+def get_localhost_instance_tags() -> Dict[str, Any]:
+    """Get current manager tags.
+
+    Returns:
+        A ``dict`` of tags (name -> value). Empty if no tags found or can't access the inst id.
+    """
+    instanceid = get_localhost_instance_id()
+    rootLogger.debug(instanceid)
+
+    resptags: Dict[str, Any] = {}
```

Comment:
same re: Dict[str, str]

Reply:
I think so but I'll put that in a later PR.